### DISCUSSION TITLE
coverage-0.34 vs coverage-1.1 project prefix and src file exclusion

### DISCUSSION THREAD

**Discussion by adrianovieira** [vitest-dev/vitest#4850] [General]
coverage-1.1 differ from coverage-0.34 showing project prefix and also ignoring src/file exclusion.

```shell
vite-project (root)
├── DRAFT
│   └── main.ts
├── index.html
├── package.json
├── package-lock.json
├── public
│   └── vite.svg
├── README.md
├── src
│   ├── App.css
│   ├── App.tsx
│   ├── assets
│   │   └── react.svg
│   ├── index.css
│   ├── main.tsx
│   └── vite-env.d.ts
├── tests
│   └── unit
│       └── basico.test.ts
├── tsconfig.json
├── tsconfig.node.json
├── vite.config.ts
└── vitest.config.js
```

- vitest.config.js  
  ```typescript
  import { configDefaults, defineConfig, mergeConfig } from 'vitest/config'
  import viteConfig from './vite.config'

  export default mergeConfig(
          viteConfig,
          defineConfig({
              test: {
                  globals: true,
                  environment: 'jsdom',
                  coverage: {reporter: ['lcov', 'html'],},
                  exclude: [
                      ...configDefaults.exclude,
                      '**/DRAFT/**/*.*',
                  ],
              },
          }),
      )
  ```

- upgraded to vitest-1.1.x

| vitest-0.34 | vitest-1.1.x |
|---|---|
|![image](https://github.com/vitest-dev/vitest/assets/5561794/213953f3-feaf-4a9b-a13e-d4f1657b2e6b)| ![image](https://github.com/vitest-dev/vitest/assets/5561794/956926f5-28cb-4e5d-abc1-1b279c52b8b8)|

As you can notice.. we get project prefix (`vite-project`) and that `DRAFT` path which is on exclude clause.

What I need to do in order to get the same behaviour from 0.34.x?

**Comment by AriPerkkio** [comment]
`coverage.all` includes all files - even the untested ones into code coverage report. Specify the file in `test.coverage.exclude` instead of `test.exclude` and it will disappear from report. Or set `coverage.all: false`.

**Reply by adrianovieira** [reply]
Nice. thx.

**Comment by adrianovieira** [comment]
Another issue I notice is that coverage show the root path of the project, e.g. `vite-project/src`.

What can I do in order to hide that name and get only the sub-folders list, e.g `src`?

**Reply by AriPerkkio** [reply]
That's handled by istanbul reporeter. You might want to look into https://github.com/istanbuljs/istanbuljs/tree/master/packages/istanbul-reports/lib/html.

I think it's showing the root directory since there are files included in report outside `src` as well.

**Comment by adrianovieira** [comment]
> I think it's showing the root directory since there are files included in report outside `src` as well.

Yes. That's it.

Thank you so much.
